django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
In PR #8432, the goal is to replace the pattern `set(<generator expression>)` with set comprehension. As this pattern occasionally creeps in over time, regular maintenance may be required keep the preferred comprehension syntax. To solve this "over time" problem, @charettes suggested the possibility of using the static analysis tool [flake8-comprehensions](https://pypi.python.org/pypi/flake8-comprehensions/1.2.1) to find and report these patterns for us. One snag with using the tool, it flagged migration files as having suboptimal set syntax `set([...])` instead of set literals.

So the goal of this change is to make the `SetSerializer` friendlier for static analysis.